api/lib/domain/usecases/update-user-details-for-administration.js

```diff
@@ -1,13 +1,39 @@
+const has = require('lodash/has');
+const {
+  AlreadyRegisteredEmailAndUsernameError,
+  AlreadyRegisteredEmailError,
+  AlreadyRegisteredUsernameError,
+} = require('../errors');
+
 module.exports = async function updateUserDetailsForAdministration({
   userId,
   userDetailsForAdministration,
   userRepository,
 }) {
-  if (userDetailsForAdministration.email) {
-    await userRepository.isEmailAllowedToUseForCurrentUser(userId, userDetailsForAdministration.email);
-  }
+  const { email, username } = userDetailsForAdministration;
+
+  const foundUsersWithEmailAlreadyUsed = email && await userRepository.findAnotherUserByEmail(userId, email);
+  const foundUsersWithUsernameAlreadyUsed = username && await userRepository.findAnotherUserByUsername(userId, username);
```

Comment:
**question**
Cette vérification avant d'appeler le repo me semble un peu smelly ... 
Si on a pas l'email ou le username on veut faire quoi ? Juste passer l'appel du répo ou jeter une erreur ?

Reply:
Ici, je n'ai modifié que le Prénom, mais tout l'object est envoyé
![image](https://user-images.githubusercontent.com/56302715/119135833-3b910d80-ba3f-11eb-9dce-82a2b31b9722.png)

```
{
	"data": {
		"attributes": {
			"cgu": false,
			"email": null,
			"first-name": "Boris",
			"last-name": "Baratheon",
			"pix-certif-terms-of-service-accepted": false,
			"pix-orga-terms-of-service-accepted": false,
			"username": "username123"
		},
		"id": "100003",
		"relationships": {
			"authentication-methods": {
				"data": [
					{
						"id": "100004",
						"type": "authentication-methods"
					}
				]
			}
		},
		"type": "users"
	}
}
```
Par contre, côté API, seules les valeurs "autorisées à la mise à jour " sont passés dans le paramètre d'entrée `userDetailsForAdministration`
http://github.com/1024pix/pix/blob/dev/api/lib/infrastructure/serializers/jsonapi/user-details-for-admin-serializer.js#L26-L34

De celles-ci, seules `email` et `username` sont contrôlées.
Donc dans le cas présent, `foundUsersWithEmailAlreadyUsed` sera `undefined` et la fonction  `_checkEmailAndUsernameAreAvailable` ne lèvera pas d'erreur

Reply:
D'accord, peut être qu'on pourrait mettre un `if` plus explicite dans ce cas ? 🤔

Reply:
Message à @manux-pix : doit-on 
- bloquer le mise à jour d'un champ non contrôlé (ex: prénom)
- si un champ contrôté (ex: email) est incorrect ?